pkg/clusteragent/admission/status.go

```diff
@@ -0,0 +1,90 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+// +build kubeapiserver
+
+package admission
+
+import (
+	"fmt"
+	"hash/fnv"
+	"strconv"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/util/kubernetes/apiserver/common"
+	"github.com/DataDog/datadog-agent/pkg/util/kubernetes/certificate"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+// GetStatus returns status info for the secret and webhook controllers.
+func GetStatus(apiCl kubernetes.Interface) map[string]interface{} {
+	status := make(map[string]interface{})
+	if !config.Datadog.GetBool("admission_controller.enabled") {
+		status["Disabled"] = "The admission controller is not enabled on the Cluster Agent"
+		return status
+	}
+
+	ns := common.GetResourcesNamespace()
+	webhookName := config.Datadog.GetString("admission_controller.webhook_name")
+	secretName := config.Datadog.GetString("admission_controller.certificate.secret_name")
+	status["WebhookName"] = webhookName
+	status["SecretName"] = fmt.Sprintf("%s/%s", ns, secretName)
+
+	webhook, err := apiCl.AdmissionregistrationV1beta1().MutatingWebhookConfigurations().Get(webhookName, metav1.GetOptions{})
```

Comment:
each time we ask for a status it will trigger a get on the API server.

Reply:
Understood

it's just if the api server didn't answer at this moment it means we may return an error.

Reply:
that's true, it's a risk we're taking for the external metrics server section as well